### PR TITLE
[cmake] disable `OPENTHREAD_CONFIG_NCP_HDLC_ENABLE` when `OT_NCP_SPI` is set

### PR DESCRIPTION
**Pull request by parag-silabs**
This change is needed to enable SPI vs UART NCP interface.

### CODE REVIEW

**Review by romacdon — APPROVED** [review]
LGTM

**Review by suveshpratapa — APPROVED** [review]
LGTM

**Review by lmnotran — APPROVED** [review]

**Comment by parag-silabs** [comment]
@jwhui Can you please review the PR?

File: src/ncp/radio.cmake [hunk 1]
```diff
@@ -47,8 +52,11 @@ target_link_libraries(openthread-rcp
...
         openthread-radio
     PRIVATE
-        openthread-hdlc
         openthread-spinel-rcp
         ot-config-radio
         ot-config
 )
+
+if(NOT OT_NCP_SPI)
+    target_link_libraries(openthread-rcp PRIVATE openthread-hdlc)
```
Comment (hemanth-silabs) [inline src/ncp/radio.cmake:61]: @parag-silabs dont we need to link libraries for SPI?

**Review by hemanth-silabs — APPROVED** [review]
Based on the private conversation I had with @parag-silabs, this PR LGTM 👍

File: src/ncp/radio.cmake (hunk 1 above)
Comment (parag-silabs) [reply src/ncp/radio.cmake:61]: No, I was able to build and run rcp over spi.

**Comment by parag-silabs** [comment]
> @jwhui Can you please review the PR?

**Comment by parag-silabs** [comment]
@jwhui Got all the approvals. Can you please have a look at the PR?

**Review by bukepo — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**